app/javascript/react/textual_summary/icon_or_image.jsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Render icon or an image with a title
+ */
+export default class IconOrImage extends React.Component {
+  constructor(props) {
+    super(props);
+    //console.log(props);
+  }
+
+  renderIcon() {
+    // %i{:class => item[:icon], :title => item[:title]}
+    return <i className={this.props.icon} title={this.props.title} />;
```

Comment:
Not now, later when we include patternfly-react in manageiq, we can do it.

Reply:
TBH I think we should just add it, no point in writing the code twice...

Reply:
"writing the code twice..."  -- you mean the single line ;-).

I will take a look at a later point. It's not a 5 minutes job. After 5 minutes of searching I found the component in patternfly-react. Figured it's not in the storybook https://rawgit.com/patternfly/patternfly-react/gh-pages/index.html. Have to figure what that means, hopefully it does not mean it should not be used.

Then I figured it's actually 2 components: https://github.com/patternfly/patternfly-react/blob/master/src/components/Icon/Icon.js and the resulting markup seems different. So have to check if it renders that same.

All that work is to prevent writing `<i className={this.props.icon} title={this.props.title} />;` Probably not. That does not match my sense of effective work.

In this PR I am trying to start this task by moving the rendering to the client side while making sure that the textual summaries render as they did before. W/o extra changes if possible.

Later we can change the markup and/or styling. In a separate PR or at least a separate commit.

Reply:
I've added https://github.com/patternfly/patternfly-react/issues/234 to track the missing storybook.

the two components are simply because we support two type of components (patternfly icons and font-awesome)
using <i> i assume its only font-awesome?

to use it you can simply do
```js
import { Icon } from 'patternfly-react';

...
export default = () => (
  <Icon type='fa' name='react' />
);